### DISCUSSION TITLE
Very new in the programing world

### DISCUSSION THREAD

**Discussion by Smailsafire92** [ImageMagick/ImageMagick#3562] [Help]
Hi,
Im sorry if its a stupid/beginner question but i'm very new to the programing world and would like to start a website where the user is going to be  uploding of images in one format, image manipulation(not to make it bigger or smaller but to add things like lines and write measurements and even add another image on top of the image), and then downloading the image in another format.

So my question is, is imageMagick the right tool and could i use it on my own website?

**Comment by fmw42** [comment]
Imagemagick will do that stuff, other than the uploading, which should be done by PHP. You can call Imagemagick command line from PHP exec() or you can use PHP Imagick API, which uses Imagemagick. See https://www.php.net/manual/en/book.imagick.php

**Reply by Smailsafire92** [reply]
Thank you for the quick respons, is there an program/tool that i could buy or do i have to make it on my own? i ask because im not that for in to my programing and a friend and I are on a type of deadline.

Thanks for the help.

**Comment by fmw42** [comment]
Imagemagick is free. So is Imagick. Programming is done by command line or script. Scripts are simple text files.  Please clarify your question.

**Reply by Smailsafire92** [reply]
Okej, so i could use your script for commercial use for a one time fee of 100$ then i will be in contact with you soon.

Thanks for the quick respons @fmw42

**Reply by fmw42** [reply]
For novices with Imagemagick, see

http://www.imagemagick.org/discourse-server/viewtopic.php?f=1&t=9620
http://www.imagemagick.org/script/command-line-processing.php
https://imagemagick.org/script/command-line-options.php
http://www.imagemagick.org/Usage/reference.html
http://www.imagemagick.org/Usage/
https://github.com/ImageMagick/usage-markdown
https://imagemagick.org/script/porting.php#cli

**Reply by fmw42** [reply]
Sorry, I thought your message was directly to me regarding my scripts. I think you are asking about Imagemagick, which is free and so is Imagick. Programming is done by command line or script. Scripts are simple text files.  Please clarify your question.
